app/controllers/api/discussions_controller.rb

```diff
@@ -3,60 +3,52 @@ class API::DiscussionsController < API::RestfulController
   load_resource only: [:create, :update]
 
   def dashboard
-    instantiate_collection { |collection| filter_collection collection }
-    respond_with_discussions
+    instantiate_collection { |collection| collection_for_dashboard collection }
+    respond_with_collection scope: { visible_in_dashboard: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
+  end
+
+  def inbox
+    instantiate_collection { |collection| collection_for_inbox collection }
+    respond_with_collection scope: { visible_in_inbox: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
```

Comment:
I was more thinking to make the discussion know which query (or queries) it
originated from, so we don't get leakage issues. So, in the record store,
we can differentiate between discussions which were loaded via the thread
page, via notifications, via dashboard, etc.

On Mon, Aug 3, 2015 at 12:38 AM, Robert Guthrie <notifications@github.com>
wrote:

> In app/controllers/api/discussions_controller.rb
> <https://github.com/loomio/loomio/pull/2369#discussion_r36061939>:
>
> > @@ -3,60 +3,52 @@ class API::DiscussionsController < API::RestfulController
> >    load_resource only: [:create, :update]
> >
> >    def dashboard
> > -    instantiate_collection { |collection| filter_collection collection }
> > -    respond_with_discussions
> > +    instantiate_collection { |collection| collection_for_dashboard collection }
> > +    respond_with_collection scope: { visible_in_dashboard: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
> > +  end
> > +
> > +  def inbox
> > +    instantiate_collection { |collection| collection_for_inbox collection }
> > +    respond_with_collection scope: { visible_in_inbox: true }, serializer: DiscussionWrapperSerializer, root: 'discussion_wrappers'
>
> Hrrmm.. what about if you just passed has_discussion_reader rather than
> visible_on_inbox ? Are you thinking that we move the "what to display"
> logic to the server side or is this just a hint to the client about what
> can be displayed?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2369/files#r36061939>.
>